waku/waku.go

```diff
@@ -151,28 +152,37 @@ func New(cfg *Config, logger *zap.Logger) *Waku {
 	waku.filters = common.NewFilters()
 
 	// p2p waku sub-protocol handler
-	waku.protocol = p2p.Protocol{
+	waku.protocols = []p2p.Protocol{{
 		Name:    v0.Name,
 		Version: uint(v0.Version),
 		Length:  v0.NumberOfMessageCodes,
-		Run:     waku.HandlePeer,
+		Run:     waku.handlePeerV0,
 		NodeInfo: func() interface{} {
 			return map[string]interface{}{
 				"version":        v0.VersionStr,
 				"maxMessageSize": waku.MaxMessageSize(),
 				"minimumPoW":     waku.MinPow(),
 			}
 		},
+	},
+		{
+			Name:    v1.Name,
+			Version: uint(v1.Version),
+			Length:  v1.NumberOfMessageCodes,
+			Run:     waku.handlePeerV1,
+			NodeInfo: func() interface{} {
+				return map[string]interface{}{
+					"version":        v1.VersionStr,
+					"maxMessageSize": waku.MaxMessageSize(),
+					"minimumPoW":     waku.MinPow(),
+				}
+			},
+		},
```

Comment:
My suggestion of using factory pattern was based on a fundamental misunderstanding of your eventual approach. Now looking at this final implementation I can see that the version will be selected by the consuming code.

Do we want to introduce in another PR the selection functionality (the code that selects the highest mutual version between peers)?

Reply:
I think that's already handled by devp2p, it will already do that behind the scene and call the right handler (specified in the `Run` field)

Reply:
Oooh, that is fancy! Cool.